serial-settings/src/lib.rs

```diff
@@ -288,88 +496,80 @@ impl<'a, P: Platform<Y>, const Y: usize> Interface<'a, P, Y> {
-                        "Settings updated. You may need to reboot for the setting to be applied"
-                    )
-            },
+                    interface,
+                    "Set but not stored. May require store and reboot to activate."
+                )
+            }
             Err(e) => {
-                writeln!(interface, "Failed to update {key}: {e:?}")
+                writeln!(interface, "Failed to set `{key}`: {e:?}")
             }
-        }.unwrap();
+        }
+        .unwrap();
     }
 
     fn menu() -> menu::Menu<'a, Self, P::Settings> {
         menu::Menu {
         label: "settings",
         items: &[
-            &menu::Item {
-                command: "list",
-                help: Some("List all available settings and their current values."),
-                item_type: menu::ItemType::Callback {
-                    function: Self::handle_list,
-                    parameters: &[],
-                },
-            },
             &menu::Item {
                 command: "get",
-                help: Some("Read a setting_from the device."),
+                help: Some("List paths and read current, default, and stored values"),
                 item_type: menu::ItemType::Callback {
                     function: Self::handle_get,
-                    parameters: &[menu::Parameter::Mandatory {
-                        parameter_name: "item",
-                        help: Some("The name of the setting to read."),
+                    parameters: &[menu::Parameter::Optional {
+                        parameter_name: "path",
+                        help: Some("The path of the value or subtree to list/read."),
                     }]
                 },
             },
             &menu::Item {
                 command: "set",
-                help: Some("Update a a setting in the device."),
+                help: Some("Update a value"),
                 item_type: menu::ItemType::Callback {
                     function: Self::handle_set,
                     parameters: &[
                         menu::Parameter::Mandatory {
-                            parameter_name: "item",
-                            help: Some("The name of the setting to write."),
+                            parameter_name: "path",
+                            help: Some("The path to set"),
                         },
                         menu::Parameter::Mandatory {
                             parameter_name: "value",
-                            help: Some("Specifies the value to be written. Values must be JSON-encoded"),
+                            help: Some("The value to be written, JSON-encoded"),
                         },
                     ]
                 },
             },
             &menu::Item {
-                command: "save",
-                help: Some("Save settings to the device."),
+                command: "store",
+                help: Some("Store values that differ from defaults"),
```

Comment:
Could this have unintended effects if a user wanted to save their operating state, then updated firmware, and some defaults changed? Their saved settings would then implicitly change underneath, which seems odd.

Reply:
Yes. There is intentionally no way to express "i want to store this value (which happens to be the default) and not have it changed if the default changes". I had the impression that distinguishing this makes it more complicated.

Reply:
If the user specifically requests to save a default value, why wouldn't we just save it?

Reply:
Because with subtrees now it would always store everything. It would be difficult to support the auto-migration of defaults.
But I'll add a `store --force` to enable that.

Reply:
https://github.com/quartiq/stabilizer/pull/929/commits/bd722757ae41200c99fbda641768d20627df2960